File: year2/systems-design-implementation/Lab5/SwimModel/src/main/java/domain/Inscriere.java
package domain;

import utils.Entity;
import utils.Pair;

import java.io.Serializable;

public class Inscriere extends Entity<Pair<Integer, Integer>> implements Serializable {

    private Participant participant;
    private Proba proba;

    public Inscriere(Integer id_participant, Integer id_proba, Participant participant, Proba proba) {
        super(new Pair<>(id_participant, id_proba));
        this.participant = participant;
        this.proba = proba;
    }

    public Inscriere(Integer id_participant, Integer id_proba) {
        super(new Pair<>(id_participant, id_proba));
        this.participant = null;
        this.proba = null;
    }

    public Participant getParticipant() {
        return participant;
    }

    public void setParticipant(Participant participant) {
        this.participant = participant;
    }

    public Proba getProba() {
        return proba;
    }

    public void setProba(Proba proba) {
        this.proba = proba;
    }

    @Override
    public String toString() {
        return "domain.Inscriere{" +
                "participant=" + participant +
                ", proba=" + proba +
                '}';
    }
}

File: year2/systems-design-implementation/Lab4/Lab4Sharp/Domain/Inscriere.cs
using Lab2Sharp.Utils;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab2Sharp.Domain
{
    public class Inscriere : Entity<Pair<int, int>>
    {
        public Participant participant { get; set; }
        public Proba proba { get; set; }
        public Inscriere(int e1, int e2)
        {
            base.id = new Pair<int, int>(e1, e2);
            participant = null;
            proba = null;
        }
        public Inscriere(int e1, int e2, Participant pa, Proba pr)
        {
            base.id = new Pair<int, int>(e1, e2);
            participant = pa;
            proba = pr;
        }
    }
}
